pkg/storage/kube/build.go

```diff
@@ -82,13 +83,18 @@ func (s *store) GetBuilds() ([]*brigade.Build, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	podList, err := s.client.CoreV1().Pods(s.namespace).List(lo)
+	if err != nil {
+		return nil, err
+	}
+
 	buildList := make([]*brigade.Build, len(secretList.Items))
 	for i := range secretList.Items {
 		b := NewBuildFromSecret(secretList.Items[i])
-		b.Worker, err = s.GetWorker(b.ID)
-		if err != nil {
-			return buildList, err
-		}
+		// The error is ErrWorkerNotFound, and in that case, we just ignore
+		// it and assign nil to the worker.
+		b.Worker, _ = findWorker(b.ID, podList)
```

Comment:
should this check if the error is `ErrWorkerNotFound`?

Reply:
I _could_, though it is currently the only possible error.